File: backend/HearthHaven.API/Data/PartnerAssignment.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace HearthHaven.API.Data;

[Table("partner_assignments")]
public class PartnerAssignment
{
    [Key]
    [Column("assignment_id")]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public int AssignmentId { get; set; }

    [Column("partner_id")]
    public int PartnerId { get; set; }

    [Column("safehouse_id")]
    public int? SafehouseId { get; set; }

    [Column("program_area")]
    public required string ProgramArea { get; set; }

    [Column("assignment_start")]
    public DateOnly AssignmentStart { get; set; }

    [Column("assignment_end")]
    public DateOnly? AssignmentEnd { get; set; }

    [Column("responsibility_notes")]
    public string? ResponsibilityNotes { get; set; }

    [Column("is_primary")]
    public bool IsPrimary { get; set; }

    [Column("status")]
    public required string Status { get; set; }

    [ForeignKey(nameof(PartnerId))]
    public Partner? Partner { get; set; }

    [ForeignKey(nameof(SafehouseId))]
    public Safehouse? Safehouse { get; set; }
}

File: frontend/src/types/PartnerAssignment.ts
import { Partner } from './Partner';
import { Safehouse } from './Safehouse';

export interface PartnerAssignment {
  assignmentId: number;
  partnerId: number;
  safehouseId: number | null;
  programArea: string;
  assignmentStart: string;
  assignmentEnd: string | null;
  responsibilityNotes: string | null;
  isPrimary: boolean;
  status: string;
  partner?: Partner;
  safehouse?: Safehouse;
}
